### PR TITLE
Introduce tenant domain/ organizationId based entity maps

### PR DESCRIPTION
**Pull request by menakaj**
### Purpose
The internal maps which are being used in Choreo Connect use, VHost:apiName:apiVersion as the map key. This have become an issue when multiple organizations having apis with the same name and version.

To fix this issue, this PR introduces, organizationID/ tenantId as the map key. 

Methodology:
1. The tenant domain parameter in API-M event is used as the organization Id to store the resources
2. In API-CTL scenario, the OrganizationId can be provided in the api.yaml file.
3. If the orgId is not provided (either in apictl mode or with api-manager) the authenticated user's tenant domain will be set as the organizationId.

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2041

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Comment by Rajith90** [comment]
Could you please Update the docs accordingly as well

File: adapter/internal/discovery/xds/server.go [hunk 1]
```diff
@@ -882,129 +930,129 @@ func UpdateEnforcerThrottleData(throttleData *throttle.ThrottleData) {
...
-			// Successful message, set the current state of the enforcer apis to the state cache.
-			buildAndStoreSuccessState(requestEvent.Node, requestEvent.Version)
-
-			// Generate the router resources and update the router.
-			listeners, clusters, routes, endpoints, _ := GenerateEnvoyResoucesForLabel(requestEvent.Node)
-			UpdateXdsCacheWithLock(requestEvent.Node, endpoints, clusters, routes, listeners)
-		}
-	}
-}
+// // buildAndStoreSuccessState method builds the latest successful enforcer and router resources and returns as
```
Comment (praminda) [inline adapter/internal/discovery/xds/server.go:933]: why commented?

Comment (menakaj) [reply adapter/internal/discovery/xds/server.go:933]: This code is not being executed. Will remove this

File: adapter/config/types.go [hunk 2]
```diff
@@ -355,6 +355,7 @@ type controlPlane struct {
 	EnvironmentLabels       []string      `toml:"environmentLabels"`
 	RetryInterval           time.Duration `toml:"retryInterval"`
 	SkipSSLVerification     bool          `toml:"skipSSLVerification"`
+	OperationMode           string        `toml:"operationMode"`
```
Comment (Rajith90) [inline adapter/config/types.go:358]: Lets remove this

File: adapter/internal/discovery/xds/consul.go [hunk 3]
```diff
@@ -35,14 +35,14 @@ const (
 	transportSocketName = "envoy.transport_sockets.tls"
 )
 
-func startConsulServiceDiscovery() {
-	for apiKey, clusterList := range openAPIClustersMap {
+func startConsulServiceDiscovery(organizationID string) {
```
Comment (Rajith90) [inline adapter/internal/discovery/xds/consul.go:38]: Where is this method called, We should call with org Id since the method signature has changed right?

**Review by Rajith90 — CHANGES_REQUESTED** [review]

File: adapter/config/types.go (hunk 2 above)
Comment (menakaj) [reply adapter/config/types.go]: Removed

File: adapter/internal/discovery/xds/consul.go (hunk 3 above)
Comment (menakaj) [reply adapter/internal/discovery/xds/consul.go:38]: Yes. This is being called when the api is added. https://github.com/wso2/product-microgateway/pull/2042/files#diff-0bb37b8dc28a8ae4fbd9179a4b374745a9868b20fa545baa063548dfe63294b6R342

**Review by praminda — APPROVED** [review]

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**